R/get_draws_df.R

```diff
@@ -5,38 +5,95 @@
 #' and the 3 relevant mappings from stan indices to the real data, in order
 #' to generate a dataframe containing the posterior draws of the counts (e.g.
 #' hospital admissions), the wastewater concentration values, the "global" R(t),
-#' and the "local" R(t) estimates + the critical metadata in the data
+#' and the "local" R(t) estimates + the critical metadata in the data.
+#' This funtion has a default method that takes the two sets of input data,
+#' the last of stan arguments, and the CmdStan fitting object, as well as an S3
+#' method for objects of class 'wwinference_fit'
 #'
 #'
-#' @param ww_data A dataframe of the preprocessed wastewater concentration data
-#' used to fit the model
+#' @param x Either a dataframe of wastewater observations, or an object of
+#' class wwinference_fit
 #' @param count_data A dataframe of the preprocessed daily count data (e.g.
 #' hospital admissions) from the "global" population
+#' @param stan_data_list A list containing all the data passed to stan for
+#' fitting the model
 #' @param fit_obj a CmdStan object that is the output of fitting the model to
-#' the `ww_data` and `count_data`
-#' @param date_time_spine A tibble mapping the time index in stan (observed +
-#' nowcast + forecast) to real dates
-#' @param lab_site_spine A tibble mapping the site-lab index in stan to the
-#' corresponding site, lab, and site population
-#' @param subpop_spine A tibble mapping the site index in stan to the
-#' corresponding subpopulation (either a site or the auxiliary site we add to
-#' represent the rest of the population)
-#'
+#' `x` and `count_data`
+#' @param ... additional arguments
 #' @return  A tibble containing the full set of posterior draws of the
 #' estimated, nowcasted, and forecasted: counts, site-level wastewater
 #' concentrations, "global"(e.g. state) R(t) estimate, and the  "local" (site +
 #' the one auxiliary subpopulation) R(t) estimates. In the instance where there
 #' are observations, the data will be joined to each draw of the predicted
 #' observation to facilitate plotting.
 #' @export
-get_draws_df <- function(ww_data,
-                         count_data,
-                         fit_obj,
-                         date_time_spine,
-                         lab_site_spine,
-                         subpop_spine) {
+get_draws_df <- function(x, ...) {
+  UseMethod("get_draws_df")
+}
+
+#' S3 method for extracting posterior draws alongside data for a
+#' wwinference_fit object
+#'
+#' This method overloads the generic get_draws_df function specifically
+#' for objects of type 'wwinference_fit'.
+#'
+#' @rdname get_draws_df
+#' @export
+get_draws_df.wwinference_fit <- function(x, ...) {
+  get_draws_df.data.frame(
+    x = x$raw_input_data$input_ww_data,
+    count_data = x$raw_input_data$input_count_data,
+    stan_data_list = x$stan_data_list,
+    fit_obj = x$fit
+  )
+}
+
+#' @export
+#' @rdname get_draws_df
+get_draws_df.default <- function(x, ...) {
```

Comment:
When would this result in an error?

Reply:
it should do so if called on anything that isn't a dataframe (or inherits from one) or a `wwinference_fit` object (and if we don;'t we should have a test to check this)